### PR TITLE
Add Comment Section

### PR DESCRIPTION
**Pull request by cbrothers8**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -23,10 +26,34 @@
...
+    public List<String> messages;
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Ciara!");
-  }
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("text/html;");
```
Comment (weiwong-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:33]: is this needed? As you are setting content type to application/json below.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -23,10 +26,34 @@
...
+
+        String json = convertToJsonUsingGson(messages);
+
+        response.setContentType("application/json;");
+        response.getWriter().println(messages);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String comment = request.getParameter("box");
```
Comment (weiwong-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:44]: "box" and "comment" etc are too generic. In this situation it is okay. However, in real world, please use more descriptive names to enhance readability.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (cbrothers8) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:33]: No, it's not needed. I forgot to remove it.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -23,10 +26,31 @@
...
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Ciara!");
-  }
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String json = convertToJsonUsingGson(messages);
+
```
Comment (weiwong-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: We are supposed to put the messages in datastore, right? Jus for troubleshooting purpose this is fine. But we still need to hook up to datastore.

Comment (cbrothers8) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: Yes, that's the next thing I have to do.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -23,10 +26,31 @@
...
+    public void init(){
+        messages = new ArrayList<>();
+    }
+
+    private String convertToJsonUsingGson(List<String> messages) {
+        Gson gson = new Gson();
+        String json = gson.toJson(messages);
+        return json;
+    }
```
Comment (weiwong-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:55]: This logic here is too simple to worth being a method.  Perhaps make it " fetch data from datastore, sort and return comments".

**Pull request merged**